vector/src/main/java/im/vector/app/features/permalink/PermalinkUseCase.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.permalink
+
+import org.matrix.android.sdk.api.session.Session
+import javax.inject.Inject
+
+/**
+ * Contains synchronous methods to create permalinks from the Session.
+ */
+class PermalinkUseCase @Inject constructor(
```

Comment:
Thats a bit weird, a UseCase should only offer one method, as it's supposed to encapsulate one business rule

Reply:
Yes, this is not really a UseCase, rather a Proxy. Which name do you suggest?